src/core/simulation_results.jl

```diff
@@ -118,26 +116,45 @@ IS.get_parameters(result::StageResults) = result.parameter_values
 #IS.get_total_cost(result::StageResults) = result.total_cost
 #IS.get_optimizer_log(results::StageResults) = results.optimizer_log
 
+_get_containers(x::StageResults) = (x.variable_values, x.parameter_values, x.dual_values)
+_get_dicts(res::StageResults) = (y for x in _get_containers(res) for y in values(x))
+
+function _get_store_value(
+    res::StageResults,
+    field::Symbol,
+    names::Vector{Symbol},
+    timestamps,
+    ::Nothing,
+)
+    simulation_store_path = joinpath(get_execution_path(res), "data_store")
+    return h5_store_open(simulation_store_path, "r") do store
+        _get_store_value(res, field, names, timestamps, store)
+    end
+end
+
 function _get_store_value(
     res::StageResults,
     field::Symbol,
     names::Vector{Symbol},
     timestamps,
+    store::SimulationStore,
 )
     results = Dict{Symbol, SortedDict{Dates.DateTime, DataFrames.DataFrame}}()
     stage_name = Symbol(get_stage_name(res))
-    simulation_store_path = joinpath(get_execution_path(res), "data_store")
     stage_interval = get_interval(res)
-    h5_store_open(simulation_store_path, "r") do store
-        for name in names
-            _results = SortedDict{Dates.DateTime, DataFrames.DataFrame}()
-            for ts in timestamps
-                out = read_result(DataFrames.DataFrame, store, stage_name, field, name, ts)
-                _results[ts] = out
-            end
-            results[name] = _results
+    resolution = PSY.get_time_series_resolution(get_system(res))
+    horizon = PSY.get_forecast_horizon(get_system(res))
+    for name in names
+        _results = SortedDict{Dates.DateTime, DataFrames.DataFrame}()
+        for ts in timestamps
+            out = read_result(DataFrames.DataFrame, store, stage_name, field, name, ts)
+            time_col = range(ts, length = horizon, step = resolution)
+            DataFrames.insertcols!(out, 1, "DateTime" => time_col)
```

Comment:
This means that the timestamps will be stored in-memory if the user calls `load_results!`.